rnd/autogpt_server/autogpt_server/server/integrations.py

```diff
@@ -84,83 +85,44 @@ async def callback(
     )
 
 
-@integrations_api_router.get(
-    "/{provider}/credentials", response_model=list[CredentialsMetaResponse]
-)
+class CredentialsResponse(CredentialsMetaResponse):
+    user_email: str
+    scopes: list[str] | None
+
+
+@integrations_api_router.get("/{provider}/credentials")
 async def list_credentials(
     provider: Annotated[str, Path(title="The provider to list credentials for")],
-    user_id: Annotated[str, Depends(get_user_id)],
+    auth: Annotated[dict[str, Any], Depends(auth_middleware)],
     store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
 ) -> list[CredentialsMetaResponse]:
-    credentials = store.get_creds_by_provider(user_id, provider)
+    credentials = store.get_creds_by_provider(auth["sub"], provider)
     return [
-        CredentialsMetaResponse(credentials_id=cred.id, credentials_type=cred.type)
+        CredentialsResponse(
+            credentials_id=cred.id,
+            credentials_type=cred.type,
+            user_email=auth["email"],
```

Comment:
I meant the e-mail linked to the credentials, not the application user e-mail. Getting such metadata for credentials is different for every service, but for e.g. Notion it's stored in the credentials' metadata.